design.md

```diff
@@ -0,0 +1,56 @@
+# Rack CLI Design Document
+
+This document will attempt to outline the design for the Rackspace Global Command Line Interface. It will not exhaustively define each interface; rather, it will focus on the philosophy and representative examples that should be used to inform the successive specifications for the specific service implementations.
+
+As such, there are 4 core tenets we're using to define the design, in order of priority:
+
+1. Correctness
+2. Consistency
+3. Completeness
+4. Terseness
+
+## Correctness
+
+As a top level objective, the CLI must operate correctly. It's unacceptable to deliver a non-functional tool, nor one that behaves contrary to the documented expectations. The key here is to document the expected behavior, and then ensure that the delivered tool matches those expectations.
+
+## Consistency
+
+Our ambition for consistency is to provide a model that is used throughout the command line interface. All calls take the following format:
+
+```
+rack <service> [--options] <model> <action> [--flags]
```

Comment:
If we want to go with global options (which it sounds like we do), they'll have to go after `rack` and before `service`:
`rack [--username <username> --region <region> --csv --fields <fields>] servers instance get --id <instanceID>`